service/release.go

```diff
@@ -51,11 +54,32 @@ func (s *ReleaseService) Create(
 		return model.Release{}, fmt.Errorf("getting project: %w", err)
 	}
 
-	rls, err := model.NewRelease(input, projectID, authorUserID)
+	// Fail early by first calling functions that depend only on user input to validate it.
```

Comment:
Valid intention, but even without nil dereference, had to spend too much time reading this method.
That's much more costly than a couple microseconds of computing.
So, I'd definitely prioritize readability and smoothness, for example something hierarchical like:
1. Organization: GitHub integration is set + get GitHub token = sort of `githubGuard`
2. Project: GitHub repo is set -> sort of `githubRepoGuard`
3. Release: GitHub tag exists -> sort of `githubTagGuard` 😄 🤷‍♀️